mm/2s2h/BenGui/SearchableMenuItems.h

```diff
@@ -0,0 +1,1514 @@
+                break;
+            }
+            auto window =
+                Ship::Context::GetInstance()->GetWindow()->GetGui()->GetGuiWindow(widget.widgetOptions.windowName);
+            if (!window) {
+                std::string msg = fmt::format("Error drawing window contents: windowName {} does not exist",
+                                              widget.widgetOptions.windowName);
+                SPDLOG_ERROR(msg.c_str());
+                break;
+            }
+            UIWidgets::WindowButton(widget.widgetName, widget.widgetCVar, window,
+                                    { .size = widget.widgetOptions.size, .tooltip = widget.widgetTooltip });
+            if (!window->IsVisible()) {
+                window->DrawElement();
+            }
+        } break;
+        case WIDGET_SEARCH: {
+            if (ImGui::Button("Clear")) {
+                menuSearch.Clear();
+            }
+            ImGui::SameLine();
+            menuSearch.Draw();
+            std::string menuSearchText(menuSearch.InputBuf);
+
+            if (menuSearchText == "") {
+                ImGui::Text("Start typing to see results.");
+                return;
+            }
+            ImGui::BeginChild("Search Results");
+            for (auto& [menuLabel, menuSidebar, cvar] : menuEntries) {
+                for (auto& sidebar : menuSidebar) {
+                    for (auto& widgets : sidebar.columnWidgets) {
+                        int column = 1;
+                        for (auto& info : widgets) {
+                            if (info.widgetType == WIDGET_SEARCH || info.widgetType == WIDGET_SEPARATOR ||
+                                info.widgetType == WIDGET_SEPARATOR_TEXT || info.isHidden) {
+                                continue;
+                            }
+                            std::string widgetStr = std::string(info.widgetName) + std::string(info.widgetTooltip);
+                            std::transform(menuSearchText.begin(), menuSearchText.end(), menuSearchText.begin(),
+                                           ::tolower);
+                            menuSearchText.erase(std::remove(menuSearchText.begin(), menuSearchText.end(), ' '),
+                                                 menuSearchText.end());
+                            std::transform(widgetStr.begin(), widgetStr.end(), widgetStr.begin(), ::tolower);
+                            widgetStr.erase(std::remove(widgetStr.begin(), widgetStr.end(), ' '), widgetStr.end());
+                            if (widgetStr.find(menuSearchText) != std::string::npos) {
+                                SearchMenuGetItem(info);
+                                ImGui::PushStyleColor(ImGuiCol_Text, UIWidgets::Colors::Gray);
+                                std::string origin =
+                                    fmt::format("  ({} -> {}, Clmn {})", menuLabel, sidebar.label, column);
+                                ImGui::Text(origin.c_str());
```

Comment:
`warning: format string is not a string literal (potentially insecure) [-Wformat-security]`

```suggestion
                                ImGui::Text("%s", origin.c_str());
```